test/accumulate.jl

```diff
@@ -192,6 +192,17 @@ end
     # Test that undefined kwargs are not accepted
     @test_throws MethodError AK.accumulate(+, v; init=10, dims=2, inclusive=false, bad=:kwarg)
 
+    # Test all options with bigger matrices
+    for D in [(1_000_000,3), (3,1_000_000)], dims in [1,2]
+        @testset let D = D, dims = dims
+            vh = ones(Float32, D)
+            v = array_from_host(vh)
+            s = AK.accumulate(+, v; init=0, dims)
+            sh = Array(s)
+            @test sh == accumulate(+, vh; init=0, dims)
+        end
+    end
```

Comment:
Should we also be testing the two different algorithms?

Reply:
`ScanPrefixes` used to be the default on Metal, and `DecoupledLookback` for the rest - we'd have to special-case which algorithm is run on each platform. I still think `DecoupledLookback` has better scalability (though for smaller arrays `ScanPrefixes` seems to almost always be faster), but it may be more susceptible to compilation bugs. Happy either way from my side

Reply:
Ah, @christiangnrd already [did that](https://github.com/JuliaGPU/AcceleratedKernels.jl/pull/49) - sorry for the spam, still catching up with activity in AK - thank you!